src/core/vector/qgsvectorlayertemporalproperties.cpp

```diff
@@ -443,22 +443,20 @@ QString QgsVectorLayerTemporalProperties::createFilterString( const QgsVectorLay
     {
       if ( mAccumulateFeatures )
       {
-        return QStringLiteral( "(%1 %2 %3) OR %1 IS NULL" ).arg( QgsExpression::quotedColumnRef( mStartFieldName ),
-               range.includeEnd() ? QStringLiteral( "<=" ) : QStringLiteral( "<" ),
-               dateTimeExpressionLiteral( range.end() ) );
+        return QStringLiteral( "(%1 < %2) OR %1 IS NULL" ).arg( QgsExpression::quotedColumnRef( mStartFieldName ), dateTimeExpressionLiteral( range.end() ) );
       }
       else if ( qgsDoubleNear( mFixedDuration, 0.0 ) )
       {
-        return QStringLiteral( "(%1 %2 %3 AND %1 %4 %5) OR %1 IS NULL" ).arg( QgsExpression::quotedColumnRef( mStartFieldName ),
-               range.includeBeginning() ? QStringLiteral( ">=" ) : QStringLiteral( ">" ),
+        // Working with features with a momentarily duration (so only one timestamp and duration about zero)
+        // You really need to take at least ONE limit into account (in practice mostly the beginning)
+        return QStringLiteral( "(%1 >= %2 AND %1 < %3) OR %1 IS NULL" ).arg( QgsExpression::quotedColumnRef( mStartFieldName ),
                dateTimeExpressionLiteral( range.begin() ),
-               range.includeEnd() ? QStringLiteral( "<=" ) : QStringLiteral( "<" ),
```

Comment:
we need to leave the range.includeEnd() ? QStringLiteral( "<=" ) : QStringLiteral( "<" ), logic here -- see above